presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1723,6 +1736,73 @@ public String getValidWriteIds(HiveIdentity identity, List<SchemaTableName> tabl
         }
     }
 
+    @Override
+    public long allocateWriteId(String dbName, String tableName, long transactionId)
+    {
+        try {
+            return retry()
+                    .stopOnIllegalExceptions()
+                    .run("allocateWriteId", stats.getOpenTransaction().wrap(() -> {
+                        try (ThriftMetastoreClient metastoreClient = createMetastoreClient()) {
+                            return metastoreClient.allocateWriteId(dbName, tableName, transactionId);
+                        }
+                    }));
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public void updateTableWriteId(String dbName, String tableName, long transactionId, long writeId)
+    {
+        checkArgument(transactionId > 0, "transactionId should be a positive integer, but was %s", transactionId);
+        requireNonNull(dbName, "dbName is null");
+        requireNonNull(tableName, "tableName is null");
+        checkArgument(writeId > 0, "writeId should be a positive integer, but was %s", writeId);
+        try {
+            retry()
+                    .stopOnIllegalExceptions()
+                    .run("alterTable", stats.getOpenTransaction().wrap(() -> {
+                        try (ThriftMetastoreClient metastoreClient = createMetastoreClient()) {
+                            metastoreClient.updateTableWriteId(dbName, tableName, transactionId, writeId);
+                        }
+                        return null;
+                    }));
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public void alterPartitions(String dbName, String tableName, List<Partition> partitions, long writeId)
```

Comment:
We do something similar already for `io.prestosql.plugin.hive.metastore.thrift.ThriftHiveMetastore#alterPartition`. We pass both `Partition` along with `PartitionStatistics`.

Do we need to update stats here too?